### PR TITLE
tests: add tests for FlattenColumnsBlock 

### LINKED ISSUE #125: Add test cases for FlattenColumnsBlock
Add tests to verify:
* Columns are correctly melted into long format
* value_name and var_name are used correctly
* Edge cases like missing or empty columns

### PR DESCRIPTION
**Pull request by shivchander**
- Introduced a new test suite for the FlattenColumnsBlock, covering various scenarios including basic flattening, handling of empty datasets, missing values, and invalid input.
- Verified the correct melting of columns into long format and ensured proper handling of edge cases.
- Enhanced test coverage to ensure robustness and reliability of the FlattenColumnsBlock implementation.

Resolves #125 

## Summary by CodeRabbit

- **Tests**
  - Added comprehensive tests to verify dataset flattening functionality, including handling of empty datasets, missing values, invalid input, and custom column names. These tests ensure correct output structure, error handling, and data integrity across various scenarios.

### CODE REVIEW

File: tests/blocks/utilblocks/test_flattenblock.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        var_cols=["non_existent_column"],
+        value_name="summary",
+        var_name="dataset_type",
+    )
+
+    with pytest.raises(KeyError):
+        block.generate(dataset)
+
+
+def test_flatten_columns_verify_melted_structure():
```
Comment (NikhilNayak-debug) [inline tests/blocks/utilblocks/test_flattenblock.py]: We most probably don't need this test case since order constraints are not necessary.

**Review by NikhilNayak-debug — APPROVED** [review]
LGTM!
